### PR TITLE
Prevent Opera Mini Hivemind

### LINKED ISSUE #8892: Investigate UUID support for Opera Mini
**Is your feature request related to a problem? Please describe.**
The NPM uuid package which sets the visitor identification value seems to be not supported by Opera Mini with Proxy mode.

The problem seems to come from the limited JavaScript support by Opera Mini in Proxy mode which should lead to a "fallback" which generates a fixed UUID.

**Describe the solution you'd like**
- Look through the UUID package for suspect lines of code that we may need to polyfill e.g. https://github.com/uuidjs/uuid/blob/master/src/rng.js#L3
- If we can't fix we could look at other uuid packages with better browser support

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**
Jira https://jira.dev.bbc.co.uk/browse/WSMEDIA-319

### PR DESCRIPTION
**Pull request by ryanmccombe**
Resolves https://github.com/bbc/simorgh/issues/8892

**Overall change:**
Our ATI setup attempts to save unique user IDs into cookies, so we can track the same user returning to the site.

Opera mini extreme mode renders pages using an intermediate server.

Our attempts to save cookies and track users of opera mini extreme is really just tracking that proxy service.

As all opera mini extreme users share that same service, this was causing them all to be tracked as the same user

**Code changes:**

- Stop attempting to track individual users in opera mini - all opera mini extreme users should be assumed to be unique

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by jroebu14 — APPROVED** [review]

File: src/app/lib/analyticsUtils/index.js [hunk 1]
```diff
@@ -151,6 +152,10 @@ export const getReferrer = (platform, origin, previousPath) => {
 export const getAtUserId = () => {
   if (!onClient()) return null;
 
+  // Users accessing the site on opera "extreme data saving mode" have the pages rendered by an intermediate service
+  // Attempting to track these users is just tracking that proxy, causing all opera mini visitors to have the same id
+  if (isOperaProxy()) return null;
```
Comment (pharingee) [inline src/app/lib/analyticsUtils/index.js:157]: Shouldn't we return a uuid rather than null?

**Review by pharingee — APPROVED** [review]
Great work. Just one comment

File: src/app/lib/analyticsUtils/index.js (hunk 1 above)
Comment (ryanmccombe) [reply src/app/lib/analyticsUtils/index.js:157]: When this is null, we don't pass an `idclient` to ATI

This seems to be what is expected when we're not trying to identify the user: https://developers.atinternet-solutions.com/general-en/craft-your-hit/

Comment (pharingee) [reply src/app/lib/analyticsUtils/index.js:157]: Oh ok. Thanks

**Pull request merged**